pkg/operator/apiserver/controller/apiservice/apiservice_controller_test.go

```diff
@@ -412,6 +437,89 @@ func TestDisabledAPIService(t *testing.T) {
 
 }
 
+func TestPreconditionsForEnabledAPIServices(t *testing.T) {
+	var (
+		bootstrapComplete = &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{Name: "bootstrap", Namespace: "kube-system"},
+			Data:       map[string]string{"status": "complete"},
+		}
+
+		bootstrapNoStatus = &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{Name: "bootstrap", Namespace: "kube-system"},
+			Data:       map[string]string{"noStatus": "complete"},
+		}
+	)
+
+	scenarios := []struct {
+		name                       string
+		existingBootstrapConfigMap *corev1.ConfigMap
+		existingAPIServices        []*apiregistrationv1.APIService
+		existingEndpoints          []*corev1.Endpoints
+
+		expectedStatus bool
+	}{
+		{
+			name: "EmptyAPIServices, NoBootstrapConfigMap",
+		},
+		{
+			name:                       "EmptyAPIServices, IncompleteBootstrapConfigMap",
+			existingBootstrapConfigMap: bootstrapNoStatus,
```

Comment:
Your call, but I would prefer inlining *corev1.ConfigMap literals here unless they were enormous or reused many times. That would save me as a reader from doing the lookup and mental translation between the variable identifier and the contents of the configmap.

Reply:
np, can change.